Shared/Extensions/ArticleUtilities.swift

```diff
@@ -56,6 +56,14 @@ extension Article {
 		return nil
 	}
 	
+	var preferredURL: URL? {
+		// Replace any space characters to handle malformed links that are otherwise percent encoded but contain spaces
+		if let link = preferredLink?.replacingOccurrences(of: " ", with: "%20"), let url = URL(string: link) {
+			return url
+		}
+		return nil
+	}
+	
```

Comment:
I’m concerned with performance implications if this is called often from the timeline view.

How about this: try creating a URL from preferredLink first and returning that. If it fails, then replace " " with "%20" and try creating a URL again. This will reduce the performance issue down to just URL initing (almost all of the time).

Reply:
Great idea. And given preferredURL is only being called in places where a URL was already being inited previously, there will be no new performance effect at all where the link is well formed.

Reply:
Revised as suggested.